crates/transaction-pool/src/pool/pending.rs

```diff
@@ -325,10 +350,58 @@ impl<T: TransactionOrdering> PendingPool<T> {
         id
     }
 
-    /// Removes the worst transaction from this pool.
-    pub(crate) fn pop_worst(&mut self) -> Option<Arc<ValidPoolTransaction<T::Transaction>>> {
-        let worst = self.all.iter().next().map(|tx| *tx.transaction.id())?;
-        self.remove_transaction(&worst)
+    /// Truncates the pool to the given limit.
+    ///
+    /// Removes transactions from the pending pool
+    /// The algorithm tries to reduce transaction counts by an approximately
+    /// equal number for all for accounts with many pending transactions.
+    pub(crate) fn truncate_pool(
```

Comment:
what do we want to do here exactly? not obvious to me what this means

Reply:
this will evict the highest nonce transactions for each unique sender, then the second highest nonce, continuing until either there are only local transactions, or the pool is under its limits. The original algorithm in geth had a lot of logic related to accounts that were over `max_account_slots`, which is not applicable because of how we validate transactions. I can document this better